### ISSUE TITLE
Failed to start WPA supplicant

### ISSUE THREAD

**Issue by purplebar0** [vanilla-wiiu/vanilla#108]
After updating to the latest continuous build, I can't sync or connect to the console any more. (Edit: This occurs even if I use the build I previously used.)

The most concerning line in the log is `FAILED TO START WPA SUPPLICANT`, so it seems that something regarding `wpa_supplicant` was broken in the meantime.

Using Steam Deck OLED under CachyOS Handheld Edition

**Comment by purplebar0** [comment]
Never mind, I don't know what is happening. I found the artifact of the build from Nov 9th and used it, but I still have this issue.

Maybe this should become a discussion.

**Comment by purplebar0** [comment]
After a bit more guesswork, I found that this isn't an issue in this build:

https://github.com/vanilla-wiiu/vanilla/actions/runs/11717832653

If I had to guess, it looks like #99 broke it.

**Comment by purplebar0** [comment]
> > If I had to guess, it looks like #99 broke it.
> 
> @purplebar0 I believe that hasn't touched anything related to vanilla-pipe, which is the component at fault here. Can you run `systemctl status iwd` so I know if iwd is running and interfering with things?

It's inactive/dead, it isn't running so it can't be interfering with anything.

**Comment by jonas04anderberg** [comment]
Anything new here? I am not using the Steam Deck but I seem to have the same issue. Or maybe this is a different issue? When I try to sync with my Wii U it just gets stuck at the sync screen and never syncs with the Wii U. When i click cancel, the log just says `FAILED TO START WPA SUPPLICANT`. My complete terminal log looks like this:

```
INFO SUCCESSFULLY BOUND SOCKET 55 on PORT 51001
RECEIVED SYNC/CONNECT SIGNAL
TEMPORARILY SET wlp1s0 TO UNMANAGED
USING WPA CONFIG: /root/.vanilla/vanilla_wpa_key.conf
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
SUBPROCESS @����w
RECEIVED UNBIND SIGNAL
SUBPROCESS @����w
FAILED TO START WPA SUPPLICANT
SETTING wlp1s0 BACK TO MANAGED
```

**Comment by anden04** [comment]
> Hi! I'm having the same issue on an AX200 on the latest version of Ubuntu, with the latest continuous build. My Wii U detects the GamePad and it says that it's going to check for updates, but Vanilla never connects to it nor syncs. For me, running vanilla-gui directly doesn't fix it

@Zagusan Yep, I’m using an AX200 on the latest version of Ubuntu too. Running vanilla-gui directly did not change anything. When you say that your Wii U detects the gamepad, do you mean that your Wii U detects that Vanilla is trying to connect or do you mean the original gamepad? For me my Wii U is not even detecting that I’m trying to connect to it.

**Comment by Zagusan** [comment]
> 
> @Zagusan Yep, I’m using an AX200 on the latest version of Ubuntu too. Running vanilla-gui directly did not change anything. When you say that your Wii U detects the gamepad, do you mean that your Wii U detects that Vanilla is trying to connect or do you mean the original gamepad? For me my Wii U is not even detecting that I’m trying to connect to it.

It detects that vanilla is trying to connect and says that it's going to check for updates to the GamePad's firmware, but vanilla never actually syncs nor connects

**Comment by coeursf** [comment]
> > @Zagusan Yep, I’m using an AX200 on the latest version of Ubuntu too. Running vanilla-gui directly did not change anything. When you say that your Wii U detects the gamepad, do you mean that your Wii U detects that Vanilla is trying to connect or do you mean the original gamepad? For me my Wii U is not even detecting that I’m trying to connect to it.
> 
> It detects that vanilla is trying to connect and says that it's going to check for updates to the GamePad's firmware, but vanilla never actually syncs nor connects

Hi, are you using self compile version or prebuild? I compile it from latest, and tried multiple distro, although there's no video, the control works good. In my case, with keyboard and [SwipSwapMe](https://github.com/wiiu-env/SwipSwapMe), will be all i need for account setting. :)

**Comment by anden04** [comment]
> > > @Zagusan Yep, I’m using an AX200 on the latest version of Ubuntu too. Running vanilla-gui directly did not change anything. When you say that your Wii U detects the gamepad, do you mean that your Wii U detects that Vanilla is trying to connect or do you mean the original gamepad? For me my Wii U is not even detecting that I’m trying to connect to it.
> > 
> > 
> > It detects that vanilla is trying to connect and says that it's going to check for updates to the GamePad's firmware, but vanilla never actually syncs nor connects
> 
> Hi, are you using self compile version or prebuild? I compile it from latest, and tried multiple distro, although there's no video, the control works good. In my case, with keyboard and [SwipSwapMe](https://github.com/wiiu-env/SwipSwapMe), will be all i need for account setting. :)

I'm using a self compile version. I have not tried a prebuild version. Maybe I should try a prebuild version and see if it makes any difference. Will let you know when I've had time to try it.

**Comment by Zagusan** [comment]
@coeursf I'm using the pre-built version. I thought it was up to date, but now that I verify it again it's not. Although I did try compile an older version of the codebase and I got some weird error about it not being able to copy a file from one place to another. I'll update you once I try with the latest code

**Comment by Zagusan** [comment]
@coeursf I have now built the latest (non pre-built) version, and vanilla is now able to sync to the console succesfully when running it directly from the compile folder, but when I try to connect there's this error:
```
SUBPROCESS execve (/home/santiago/Downloads/vanilla-master/build/bin/../sbin/dhcp.sh, ...): Permission denied
```
It never actually manages to connect, even though I ran the vanilla-gui as sudo

This is probably getting a bit out of topic though, so if I can't fix it I'll create another separate issue. Also, thanks for your help by the way!

**Comment by Zagusan** [comment]
> instead of install it by `cmake --install .` i run the vanilla-gui directly in the build/bin/ this magically sovled this and paired, but i still got no video, i'm using ax201. hope this information helps :)

Also I found out this happens because when installing, it doesn't seem to copy `dhcp.sh` so it can't find it:
```
SUBPROCESS execve (/usr/local/bin/../sbin/dhcp.sh, ...): No such file or directory
```

**Comment by Boidaboi** [comment]
Hi. Don't really want ot waste anybody's time but I'm having issues getting my steam deck to connect to my Wii u. It just sits on subprocess, I've tried everything I can find and can't seem to get anything to work, thought this most fits in in this thread.

**Comment by Zagusan** [comment]
> Hi. Don't really want ot waste anybody's time but I'm having issues getting my steam deck to connect to my Wii u. It just sits on subprocess, I've tried everything I can find and can't seem to get anything to work, thought this most fits in in this thread.

@Boidaboi hi there! I don't have a Steam Deck, so I can't be of much help, but I've seen online some threads talking about this exact problem on. They mention something about switching from `iw` to `wpa_supplicant` on the developer settings

**Comment by theotheroracle** [comment]
i'm also having this issue on linux mint, and i can confirm that it works before #99, and does not work after #99, so as far as i can tell this is the issue

**Comment by theotheroracle** [comment]
i get this when i use a ci build or if i cmake --install the binaries, i noticed it was modifying the binaries before installing them but i dont know cmake well enough to figure out what it's doing, but when i try to pair it just hangs forever, and when i cancel, i get failed to start wpa supplicant . doesn't happen if i run build/bin/vanilla-gui directly though

**Comment by itsmattkc** [comment]
Vanilla has gone through some major overhauls recently and should be significantly more stable and reliable. Please try again with the latest build.

**Comment by purplebar0** [comment]
Yep, that's no longer an issue.
